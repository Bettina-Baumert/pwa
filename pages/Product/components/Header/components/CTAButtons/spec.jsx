/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

const mockedState = {
  product: {
    currentProduct: {
      productId: 'foo',
    },
    productsById: {
      foo: {
        isFetching: false,
        productData: {
          id: 'foo',
          name: 'Foo Name',
          flags: {
            hasVariants: false,
            hasOptions: false,
          },
        },
      },
    },
  },
  favorites: {
    products: {
      isFetching: false,
      ids: ['foo'],
    },
  },
};

describe('CTAs (product header)', () => {
  /**
   * Creates component
   * @return {ReactWrapper}
   */
  const createComponent = () => {
    const mockedStore = configureStore();

    /* eslint-disable global-require */
    const CTAButtons = require('./index').default;
    /* eslint-enable global-require */
    return mount(
      <Provider store={mockedStore(mockedState)}>
        <CTAButtons handleAddToCart={() => {}} />
      </Provider>
    );
  };

  describe('Rendering', () => {
    it('should render CTAButtons when data is available', () => {
      const component = createComponent();
      expect(component).toMatchSnapshot();
      expect(component.find('FavoritesButton').exists()).toBe(true);
      expect(component.find('AddToCartButton').exists()).toBe(true);
    });
  });
});
