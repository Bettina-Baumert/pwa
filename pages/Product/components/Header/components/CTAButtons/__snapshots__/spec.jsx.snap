// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAs (product header) should render CTAButtons when data is available 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CTAButtons)
    productId="product_1"
  >
    <CTAButtons
      addToCart={[Function]}
      isDisabled={true}
      isFavorite={true}
      isLoading={false}
      options={Object {}}
      productId="product_1"
    >
      <div
        className="css-7azkrx"
      >
        <Connect(FavoritesButton)
          active={true}
          className="css-j05drb"
          productId="product_1"
          rippleClassName="css-1tdptiz"
        >
          <FavoritesButton
            active={true}
            addFavorites={[Function]}
            className="css-j05drb"
            noShadow={false}
            onRippleComplete={[Function]}
            once={false}
            productId="product_1"
            removeFavorites={[Function]}
            removeThrottle={0}
            removeWithRelatives={false}
            rippleClassName="css-1tdptiz"
          />
        </Connect(FavoritesButton)>
        <AddToCartButton
          buttonSize={56}
          className={null}
          iconSize={24}
          isDisabled={true}
          isLoading={false}
          noShadow={false}
          onClick={[Function]}
        >
          <button
            className="null css-wuq2ta css-15s4nvk"
            data-test-id="addToCartButton"
            onClick={[Function]}
          >
            <div
              className="css-rukjns"
              style={
                Object {
                  "opacity": 1,
                  "transform": "translate3d(0, 300%, 0)",
                }
              }
            >
              <Tick>
                <Icon
                  className=""
                  color={null}
                  content="<path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"/>"
                  size="inherit"
                  viewBox="0 0 24 24"
                >
                  <svg
                    className="css-yys9hb "
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"/>",
                      }
                    }
                    style={
                      Object {
                        "fill": null,
                        "fontSize": "inherit",
                      }
                    }
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </Icon>
              </Tick>
            </div>
            <div
              className="css-rukjns"
              onAnimationEnd={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "transform": "translate3d(0, -50%, 0)",
                }
              }
            >
              <CartPlus>
                <Icon
                  className=""
                  color={null}
                  content="<path d=\\"M0 0h24v24H0zm18.31 6l-2.76 5z\\" fill=\\"none\\"/><path d=\\"M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z\\"/>"
                  size="inherit"
                  viewBox="0 0 24 24"
                >
                  <svg
                    className="css-yys9hb "
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<path d=\\"M0 0h24v24H0zm18.31 6l-2.76 5z\\" fill=\\"none\\"/><path d=\\"M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z\\"/>",
                      }
                    }
                    style={
                      Object {
                        "fill": null,
                        "fontSize": "inherit",
                      }
                    }
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </Icon>
              </CartPlus>
            </div>
          </button>
        </AddToCartButton>
      </div>
    </CTAButtons>
  </Connect(CTAButtons)>
</Provider>
`;
